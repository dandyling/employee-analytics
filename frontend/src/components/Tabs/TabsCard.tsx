import React, { ReactNode, useState } from "react";
import { Card } from "../card/Card";
import { CardBody } from "../card/CardBody";
import { CardHeader } from "../card/CardHeader";
import { TabBar, TabHeader } from "./TabBar";

export interface Tab extends TabHeader {
  tabContent: ReactNode;
}

interface TabsCardProps {
  title?: string;
  tabs: Tab[];
  className?: string;
}

export const TabsCard = (props: TabsCardProps) => {
  const { title, tabs, className = "" } = props;
  const [index, setIndex] = useState(0);
  const { tabContent } = tabs[index];

  const handleChange = (i: number) => {
    setIndex(i);
  };

  return (
    <Card className={className}>
      <CardHeader>
        {title && <h3 className="mr-4">{title}</h3>}
        <TabBar tabs={tabs} selected={index} onChange={handleChange} />
      </CardHeader>
      <CardBody>{tabContent}</CardBody>
    </Card>
  );
};
