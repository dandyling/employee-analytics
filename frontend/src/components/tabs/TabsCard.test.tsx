import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { Tab, TabsCard } from "./TabsCard";

const testTabs: Tab[] = [
  { tabName: "Tab 1", tabContent: <div>Tab Content 1</div> },
  { tabName: "Tab 2", tabContent: <div>Tab Content 2</div> },
];

test("renders tab headers and tab contents", () => {
  render(<TabsCard tabs={testTabs} />);
  const tab1 = screen.getByText(/Tab 1/i);
  expect(tab1).toBeInTheDocument();
  const tab2 = screen.getByText(/Tab 2/i);
  expect(tab2).toBeInTheDocument();
  const tabContent1 = screen.getByText(/Tab Content 1/i);
  expect(tabContent1).toBeInTheDocument();
});

test("changes tab when clicked", () => {
  render(<TabsCard tabs={testTabs} />);
  const tab1 = screen.getByText(/Tab 2/i);
  fireEvent.click(tab1);
  const tabContent2 = screen.getByText(/Tab Content 2/i);
  expect(tabContent2).toBeInTheDocument();
});

test("renders correctly", () => {
  const wrapper = <TabsCard tabs={testTabs} />;
  expect(wrapper).toMatchSnapshot();
});
