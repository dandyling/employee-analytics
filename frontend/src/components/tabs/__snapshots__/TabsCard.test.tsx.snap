// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TabsCard
  tabs={
    Array [
      Object {
        "tabContent": <div>
          Tab Content 1
        </div>,
        "tabName": "Tab 1",
      },
      Object {
        "tabContent": <div>
          Tab Content 2
        </div>,
        "tabName": "Tab 2",
      },
    ]
  }
/>
`;
